### PR TITLE
Glue Catalog: Basic Setup + Test Infra (1/3)

### PR DESCRIPTION
**Pull request by marvinlanhenke**
### Which issue does this PR close?
Partly #249 (Task 1/3)

### Rationale for this change
Add support for Glue catalog, to reach feature parity with other implementations.

### What changes are included in this PR?
- Setup basic structure (crate/trait)
- Setup basic infra (docker) for integration tests
- implemented `list_namespaces` to have an end-to-end use-case to verify setup
- also added `glue-catalog` to `dependencies.py` and updated those files

### Are these changes tested?
Yes. Unit and integration tests are included.

### CODE REVIEW

**Comment by marvinlanhenke** [comment]
@liurenjie1024 
PTAL

File: crates/catalog/glue/src/catalog.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+use typed_builder::TypedBuilder;
+
+use crate::error::from_aws_error;
+use crate::utils::create_sdk_config;
+
+#[derive(Debug, TypedBuilder)]
+/// Glue Catalog configuration
+pub struct GlueCatalogConfig {
+    #[builder(default, setter(strip_option))]
+    endpoint_url: Option<String>,
```
Comment (liurenjie1024) [inline crates/catalog/glue/src/catalog.rs:34]:
```suggestion
    uri: Option<String>,
```
Though not required, Java and python both follow the pattern that it's called uri, I would suggest to keep the same name pattern.

**Review by liurenjie1024 — COMMENTED** [review]
Thanks @marvinlanhenke , this pr looks great! I have left some minor suggestions.

File: crates/catalog/hms/DEPENDENCIES.rust.tsv
[the remark this answers is not in the document]
Comment (marvinlanhenke) [reply crates/catalog/hms/DEPENDENCIES.rust.tsv:1]: Yes. It was more of an accident than an update on purpose. I'll see that I exclude those from the PR.

File: crates/catalog/glue/src/catalog.rs (hunk 1 above)
Comment (marvinlanhenke) [reply crates/catalog/glue/src/catalog.rs:34]: Sure I'll change that - thanks for pointing this out.

**Comment by marvinlanhenke** [comment]
@liurenjie1024
Thanks for the review, I'll get to the minor fixes this evening - and continue the implementation (task 2/3) on another PR during next week.

File: crates/catalog/hms/DEPENDENCIES.rust.tsv
[the remark this answers is not in the document]
Comment (marvinlanhenke) [reply crates/catalog/hms/DEPENDENCIES.rust.tsv:1]: should be fine now.

**Comment by marvinlanhenke** [comment]
@liurenjie1024
applied the fixes - everything should be fine now - thanks again for the quick review.

**Review by liurenjie1024 — APPROVED** [review]
LGTM, thanks!

**Pull request merged**